File: benches/mutations/variants/python/seed5.t4-inline-helper.py
def normalize_readings(values, limit):
    cleaned = []
    for v in values:
        if v > limit:
            cleaned.append(limit)
        elif v < 0:
            cleaned.append(0)
        else:
            cleaned.append(v)
    total = 0
    for c in cleaned:
        total = total + c
    if total > 500:
        cleaned.append(total / 2)
    return cleaned

File: benches/mutations/variants/rust/seed5.t4-inline-helper.rs
pub fn normalize_readings(values: &[f64], limit: f64) -> Vec<f64> {
    let mut cleaned = Vec::new();
    for v in values {
        let adjusted = if *v > limit {
            limit
        } else if *v < 0.0 {
            0.0
        } else {
            *v
        };
        cleaned.push(adjusted * 10.0);
    }
    let mut total = 0.0;
    for c in &cleaned {
        total += *c;
    }
    if total > 500.0 {
        cleaned.push(total / 2.0);
    }
    cleaned
}
